### DISCUSSION TITLE
No access to private posts on remote instance?

### DISCUSSION THREAD

**Discussion by kr428** [pixelfed/pixelfed#5117] [Q&A] [answered]
Setup: I'm on Mastodon (or Friendica) and following a pixelfed user with a "private" profile. In this environment, I cannot see most of this contacts posts. Why?

- Looking at this user from my Mastodon (or Friendica) client, I only see the posts that reached my instance, and at some point the system says "go to remote instance to see more".
- Unfortunately, clicking on the profile link, I only see "Account is private" and pixelfed prompting me to log in.

This however, doesn't work: I don't have a pixelfed account on that platform, and even if I had, I wouldn't necessarily be following this particular account. I can't use my Mastodon or Friendica login credentials on pixelfed either. So at the moment, for private profiles like this, "old" data is inaccessible from remote.

This is a difficult situation to be left in. You see any way to handle this...?

**Comment by abanink** [comment]
This is exactly a problem that [OpenWebAuth](https://zotlabs.org/page/zot/specs+openwebauth) solves.

I even made a proof-of-concept implementation of Pixelfed as a OpenWebAuth server (authorizing access to remotely authenticated users).
Now the biggest issue is that Pixelfed doesn't have the concept of authorized users who have not logged in to the Pixelfed server. In my proof-of-concept, to simulate this, I added a constraint that the authorized user must be following the user whose post he wants to see (instead of 'This account is private') when retrieving the posts. But then all links from that page fail because they lack support for remotely authenticated users as well, that's why it's just a proof of concept and not really usable atm.

But it's a way to handle this situation.

**Comment by trwnh** [comment, marked as answer]
some form of cross-domain authorization is needed, yeah. not necessarily openwebauth, though -- you could use indieauth instead, for example. in general, you basically prove to the remote website who you are and then it shows you private stuff that you're allowed to see. but it would indeed require a lot of changes and also require agreement from the rest of the ecosystem on what to actually use to establish your identity across domains.

**Comment by kr428** [comment]
I'm not really biased towards any of these implementations (though of course also tending to think OpenWebAuth simply because it would create interoperability with Hubzilla which I deem a good thing - we have way too many incompatible solutions for such things in the Fediverse already). Unsure, however, whether it really would _require_ such a solution. The other way 'round: I can very well look at a private profile I'm following from my instance. Maybe something like "fetch all posts from the remote system" triggered from my instance would be a viable solution to that that doesn't require cross-domain authentication? Only thing I'd see here: This currently feels like sort of a "bug" (as it renders the system unusable in this particular setup) and so I guess it needs _any_ kind of solution...
